servers/src/main/java/tachyon/worker/TachyonWorker.java

```diff
@@ -15,20 +15,111 @@
 import tachyon.worker.file.FileSystemWorker;
 
 /**
  * Entry point for the Tachyon Worker. This class is responsible for initializing the different
  * workers that are configured to run.
+ * <p>
+ * This class is not thread-safe.
  */
 public final class TachyonWorker {
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
+  private TachyonConf mTachyonConf;
+  // The workers
+  /** The worker serving blocks */
+  private BlockWorker mBlockWorker;
+  /** The worker serving file system operations */
+  private FileSystemWorker mFileSystemWorker;
+
+  /** is true if the worker is serving the RPC server */
+  private boolean mIsServingRPC = false;
+  /** Worker metrics system */
+  private MetricsSystem mWorkerMetricsSystem;
+  /** Worker Web UI server */
+  private UIWebServer mWebServer;
+  /** Thread pool for thrift */
+  private TThreadPoolServer mThriftServer;
+  /** Server socket for thrift */
+  private TServerSocket mThriftServerSocket;
+  /** RPC local port for thrift */
+  private int mRPCPort;
+  /** The address for the rpc server */
+  private InetSocketAddress mWorkerAddress;
+  /** Net address of this worker */
+  private WorkerNetAddress mWorkerNetAddress;
+  /** Worker start time in milliseconds */
+  private long mStartTimeMs;
+
+  /**
+   * Constructor of {@link TachyonWorker}.
+   */
+  public TachyonWorker() {
+    try {
+      mStartTimeMs = System.currentTimeMillis();
+      mTachyonConf = WorkerContext.getConf();
+
+      mBlockWorker = new BlockWorker();
+      // Setup the file worker
+      mFileSystemWorker = new FileSystemWorker(mBlockWorker.getBlockDataManager());
+
+      // Setup metrics collection system
+      mWorkerMetricsSystem = new MetricsSystem("worker", mTachyonConf);
+      WorkerSource workerSource = WorkerContext.getWorkerSource();
+      workerSource.registerGauges(mBlockWorker.getBlockDataManager());
+      mWorkerMetricsSystem.registerSource(workerSource);
+
+      // Setup web server
+      mWebServer =
+          new WorkerUIWebServer(ServiceType.WORKER_WEB, NetworkAddressUtils.getBindAddress(
+              ServiceType.WORKER_WEB, mTachyonConf), mBlockWorker.getBlockDataManager(),
+              NetworkAddressUtils.getConnectAddress(ServiceType.WORKER_RPC, mTachyonConf),
+              mStartTimeMs, mTachyonConf);
+
+      // Setup Thrift server
+      mThriftServerSocket = createThriftServerSocket();
+      mRPCPort = NetworkAddressUtils.getThriftPort(mThriftServerSocket);
+      // Reset worker RPC port based on assigned port number
+      mTachyonConf.set(Constants.WORKER_RPC_PORT, Integer.toString(mRPCPort));
+      mThriftServer = createThriftServer();
+
+      mWorkerAddress =
+          NetworkAddressUtils.getConnectAddress(NetworkAddressUtils.ServiceType.WORKER_RPC,
+              mTachyonConf);
+
+    } catch (Exception e) {
+      LOG.error("Failed to initialize {}", this.getClass().getName(), e);
+      System.exit(-1);
```

Comment:
Do we need the exception handling here, since it will be handled in the main method?